File: miscellaneous/sorting/quick-sort/20221031/QuickSortTest.php
<?php

declare(strict_types=1);

use PHPUnit\Framework\TestCase;

require './quick-sort.php';

class QuickSortTest extends TestCase
{
    private array $testCases;

    function __construct()
    {
        $this->testCases = [
            [],
            [2],
            [2, 1],
            [99, 44, 6, 2],
            [1, 2, 3, 4, 5, 6, 9, 100],
            [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0],
        ];

        parent::__construct();
    }

    function test_quick_sort()
    {
        array_map(function(array $nums) {
            $sorted = $nums;
            sort($sorted);
            $this->assertEquals($sorted, quickSort($nums, 0, count($nums) - 1 ));
        }, $this->testCases);
    }
}

File: miscellaneous/sorting/quick-sort/20221105/quick_sort_test.py
from quick_sort import quick_sort


test_cases = [
    [],
    [2],
    [2, 1],
    [99, 44, 6, 2],
    [1, 2, 3, 4, 5, 6, 9, 100],
    [99, 44, 6, 2, 1, 5, 63, 87, 283, 4, 0],
]


def test_quick_sort():
    for test_case in test_cases:
        assert quick_sort(test_case, 0, len(test_case) - 1) == sorted(test_case)
